cost/google/cud_recommendations/google_committed_use_discount_recommendations.pt

```diff
@@ -220,65 +239,129 @@ script "js_instance_cost_mapping", type:"javascript" do
       return result+"."+values[1];
     }
     total = 0
-    cur = ""
-    _.each(recommendations, function(recommendation){
+    var cur = ""
+    var separator = ","
+
+    //Create algorithm mapping (Parameter name mapped to API name & user-friendly name)
+    algorithm_mapping = {
+      "Stable Usage": {
+        "value": "LOW_WATERMARK",
+        "name": "Stable Usage (Low Watermark)"
+      },
+      "Optimal": {
+        "value": "BREAK_EVEN_POINT",
+        "name": "Optimal (Break-Even Point)"
+      }
+    }
+
+    //Filter recommendations on Algorithm parameter
+    recommendations = _.filter(recommendations, function(recommendation) {
+      return recommendation.content.overview.algorithm == algorithm_mapping[param_algorithm].value
+    })
+
+    //For each recommendation calculate savings / give algorithm user-friendly name
+    _.each(recommendations, function(recommendation) {
       count++
-      if(recommendation['currency'] !== undefined ) {
+      if ( recommendation['currency'] !== undefined ) {
         if (ds_currency_reference[recommendation['currency']] !== undefined ) {
-        var cur = ds_currency_reference[recommendation['currency']]['symbol']
+          cur = ds_currency_reference[recommendation['currency']]['symbol']
         } else {
-          var cur = "$"
+          cur = "$"
         }
         recommendation['savingsCurrency'] = cur
       } else {
-        var cur = "$"
-        var separator = ","
+        cur = "$"
+        separator = ","
+      }
+
+      //Convert Costs from Negative to Positive and add Units and Nanos
+      if (!(recommendation.costNanos == null || recommendation.costNanos == undefined)) {
+        if (recommendation.costNanos != 0) {
+          recommendation.costNanos = recommendation.costNanos * -1
+        }
+      } else {
+        recommendation.costNanos = 0
       }
-      if (recommendation.costNanos<0) {
-        recommendation.costNanos = recommendation.costNanos * -1
+
+      if (!(recommendation.costUnits == null || recommendation.costUnits == undefined)) {
+        if (recommendation.costUnits != 0) {
```

Comment:
Probably we can move this conditional to line 286 too.

Reply:
Thanks @juanestrahcl , this has been addressed.